src/components/DocumentationTopic/PrimaryContent/DeclarationGroup.vue

```diff
@@ -80,6 +147,32 @@ export default {
       return this.declaration.platforms.join(', ');
     },
     isSwift: ({ interfaceLanguage }) => interfaceLanguage === Language.swift.key.api,
+    references: ({ store }) => store.state.references,
+    isExpanded: {
+      get: ({ declListExpanded }) => declListExpanded,
+      set(value) {
+        this.$emit('update:declListExpanded', value);
+      },
+    },
+  },
+  methods: {
+    async selectDeclaration(identifier) {
+      if (identifier === this.identifier) return;
+      this.selectedIdentifier = identifier;
+      // enough time to update the just selected item
+      await waitFor(100);
+      this.isExpanded = false; // collapse the list
+      // await animation to finish
+      await waitFor(500);
```

Comment:
Is this to match the duration of a CSS transition? I imagine it's not trivial, but it would be nice if we made were able to make this programmatic in the future instead of hardcoded to match the duration.

Reply:
Ideally it should be possible to wait for a transition, its just that `transitionend` is a bit whacky and fires for child transitions as well.

Reply:
It's longer than the duration of the transition. We use `waitFor` for the sidebar animation as well. Would be nice to make both of these programmatic in the future.